doc/basics.md

```diff
@@ -399,6 +399,32 @@ And another one:
 
 Notice how we can string several operation (`parser.parse(abstract_json)["str"]["123"]["abc"].get<double>()`) and only check for the error once, a strategy we call  *error chaining*.
 
+
+The next two functions will take as input a JSON document containing an array with a single element, either a string or a number. They return true upon success.
+
+```C++
+simdjson::dom::parser parser{};
+
+bool ParseDouble(const char *j, double &d) {
+  simdjson::error_code error;
+  parser.parse(j, std::strlen(j))
+        .at(0)
+        .get<double>()
+        .tie(d, error);
+  if (error) { return false; }
+  return true;
+}
+
+bool ParseString(const char *j, std::string &s) {
+  auto [answer, error] = parser.parse(j,strlen(j))
```

Comment:
I'm sorry I brought this up. I forgot we were C++11-bound in the readme. I'd just switch this back to tie() and leave it that way. Sorry. In fact, I'mma go in and fix these in your branch so you don't have to, and you can tell *me* if you think it's OK.